### ISSUE TITLE
Tooltip in narrow window does not position correctly and causes flicker

### ISSUE THREAD

**Issue by jylertones** [twbs/bootstrap#15515]
We have one page that runs in a fairly narrow iframe. It has a simple list of links, each with their own tooltip. We initialize the tooltips with this simple configuration:

    $('[title]').tooltip({
        container: 'body',
        placement: 'auto'
    });

At large window widths the tooltips do not wrap and everything displays fine.
At smaller window widths the tooltips that have wrapping text display over the triggering link element. Because they display over the triggering link element, if you keep your cursor over the element and tooltip, it may flicker in a way that makes it appear and disappear every few seconds.

Demo: http://jsbin.com/ropubo/1/
(On the demo, try making the window about 100px wide and try the tooltips.

![image](https://cloud.githubusercontent.com/assets/1456156/5652388/fcb8a316-9664-11e4-87b4-6ab46b382e92.png)

**Comment by twbs-lmvtfy** [comment]
Hi @jylertones!

You appear to have posted a live example (http://jsbin.com/ropubo/1/edit), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* [W002](https://github.com/twbs/bootlint/wiki/W002): `<head>` is missing X-UA-Compatible `<meta>` tag that disables old IE compatibility modes
* [W003](https://github.com/twbs/bootlint/wiki/W003): `<head>` is missing viewport `<meta>` tag that enables responsiveness

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by jylertones** [comment]
Updated Demo: http://jsbin.com/ropubo/1/

**Comment by cvrebert** [comment]
@jylertones Please make a stable clone of the example; you seem to be live-editing it still.

**Comment by jylertones** [comment]
Here's a clone: http://jsbin.com/cuyubi/1

(I'm no longer touching either of them. Sorry for the trouble!)

**Comment by cvrebert** [comment]
I'm unable to reproduce with http://jsbin.com/cuyubi/1 on OS X Chrome.
@jylertones On what OS and browser are you experiencing the problem?

**Comment by jylertones** [comment]
I can get the overlap in Chrome (Canary at the moment), but for
overlap+flashing, I see it best in Firefox (34.0.5)

**Comment by cvrebert** [comment]
@jylertones *On what OS*?

**Comment by cvrebert** [comment]
I'm unable to repro on OS X Firefox 34.0.5.

**Comment by jylertones** [comment]
Sorry, I'm on OSX 10.10. Also of note, for me to reproduce this, I pretty much have to go on edit mode on  JSBin to make the window narrow enough. My browser alone won't let me make the window narrow enough.

**Comment by cvrebert** [comment]
> for me to reproduce this, I pretty much have to go on edit mode on JSBin to make the window narrow enough.

Yup, I've been doing that...
Since I can't seem to repro, could you post a screencast of the problem?

**Comment by jylertones** [comment]
Perhaps v3 of this repro is better, the config in v1 is placement: 'bottom' which seems to work

This may work better: http://jsbin.com/cuyubi/3
 
Here's a screencast (if animated gif is hard, let me know a better place to post it):
![tooltip_flicker mov](https://cloud.githubusercontent.com/assets/1456156/5655464/4bd8c85e-9683-11e4-969b-a29e621f3f8f.gif)

**Comment by cvrebert** [comment]
Okay, I'm seeing it at ~145px wide, and ironically the tooltip for the link named "Broken" works fine.
Frozen copy of v3: http://jsbin.com/qufuma/1/edit?html,output

**Comment by jh3y** [comment]
Did anyone ever find a solution to this? I'm looking for issues as I want to start contributing.

With this one though, it could be more like a browser thing than Bootstrap specific.

If you add a hover effect to the anchor, the effect will also flicker sometimes; http://jsbin.com/fizuzewita/2/edit?html,css,js,console,output

I tried this with the anchors also set to `display: block;` And although they worked, sometimes moving upwards too quickly through them meant certain tooltips weren't fired.

I'd be happy to look into this though if it's felt that this could be caught in the JavaScript. 

@jh3y

**Comment by twbs-lmvtfy** [comment]
Hi @jh3y!

You appear to have posted a live example (http://jsbin.com/fizuzewita/2/edit), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 7, column 1: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.2 ; saw what appears to be usage of Bootstrap 3.3.1
* line 8, column 1: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.2 ; saw what appears to be usage of Bootstrap 3.3.1

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by cvrebert** [comment]
@jh3y Nobody has proposed a fix yet; it's up for grabs.

**Comment by mdo** [comment]
Punting as a won't fix—v4 includes a new positioning library for our affix, tooltips, and popovers.
